sql/log.cc

```diff
@@ -2282,6 +2282,9 @@ bool File_query_log::set_rotated_name(const bool need_lock)
     }
 
     if (need_lock) mysql_mutex_lock(&LOCK_global_system_variables);
+    if (opt_slow_logname != NULL) {
+      my_free(opt_slow_logname);
```

Comment:
where do we free without rotation?

Reply:
This should be done in Sys_var_charptr::cleanup() which is invoked in sys_var_end().